core/src/tile/tileManager.cpp

```diff
@@ -369,6 +369,8 @@ bool TileManager::addTile(TileSet& tileSet, const TileID& _tileID) {
     if (tile) {
         m_tiles.push_back(tile);
 
+        // Update tile origin based on wrap (set in the new tileID)
+        tile->updateTileOrigin(_tileID.wrap);
```

Comment:
Yes. Its when a tile data (corresponding to some xyz tileID) has been cached, and a new tile is being added with the wrapped IDs same as the cached tile's IDs. In this case you would want to have the tile grab the data from the cache and update the tile origin with the new wrap.